src/main/java/org/skife/jdbi/v2/unstable/BindIn.java

```diff
@@ -90,21 +100,98 @@ public void apply(SQLStatement q) throws SQLException
     class BindingFactory implements BinderFactory<BindIn>
     {
         @Override
-        public Binder build(BindIn in)
+        public Binder build(final BindIn in)
         {
             final String key = in.value();
 
-            return new Binder<Annotation, Iterable>()
+            return new Binder<Annotation, Object>()
             {
                 @Override
-                public void bind(SQLStatement q, Annotation bind, Iterable coll)
+                public void bind(final SQLStatement q, final Annotation bind, final Object arg)
                 {
+                    final Object[] arr = Util.toArray(arg);
+
                     int idx = 0;
-                    for (Object s : coll) {
+                    for (final Object s : arr)
+                    {
                         q.bind("__" + key + "_" + idx++, s);
                     }
                 }
             };
         }
     }
+
+    final class Util
+    {
+        private Util()
+        {
+        }
+
+        static Object[] toArray(final Object obj)
+        {
+            if (obj instanceof Iterable)
+            {
+                final Iterable<?> iterable = (Iterable) obj;
+                final List<Object> out = new ArrayList<Object>();
+
+                for (final Object x : iterable)
+                {
+                    out.add(x);
+                }
+
+                return out.toArray();
+            }
+
+            if (obj.getClass().isArray())
+            {
+                if (obj instanceof Object[])
+                {
+                    return (Object[]) obj;
+                } else
+                {
+                    final int length = Array.getLength(obj);
+                    final Object[] out = new Object[length];
+
+                    for (int i = 0; i < length; i++)
+                    {
+                        out[i] = Array.get(obj, i);
+                    }
+
+                    return out;
+                }
+            }
+
+            throw new IllegalArgumentException("argument must be one of the following: Iterable, or an array/varargs (primitive or complex type)");
```

Comment:
Could consider appending `obj.getClass()` to the reason here, so you know what type was actually sent in, in case it isn't what you were expecting.